DDS-1203 cleanup local dev environment

dmlond opened the pull request:
beginning process of removing unused features, and updating our local docker environment

dmlond committed:
upgraded node to version 10.15, which is closer to the heroku npm 10.14

dmlond committed:
removing environment variables and code for local ssl. This is not necessary.
puma.cfg also had outdated puma commands for use by jruby which do not work with regular ruby

dmann reviewed (commented):
Looks good so far! :+1:

dmlond committed:
this was a spike on the ELK stack for docker log aggregation before kubernetes existed

dmlond committed:
version 1 circleci configuration no longer needed

dmlond committed:
removed references to api_test_user rake task and docker-compose.circle.yml

dmlond committed:
no longer actively using gitlab-ci, but this documentation would still work if we decide to
in the future. Changed wording slightly to reflect that we can use this, and removed
the section on the .gitlab-ci.yml file since it was removed.

dmlond committed:
no longer being used in any Dockerfile

dmlond committed:
removed one too many files from docker/includes

dmlond committed:
moved install_node.sh
uses node 10.14.2

dmlond committed:
no longer used by circle

dmlond commented:
this PR is ready for final review and merge

dmann reviewed (approved):
:shipit:
